x/avs/keeper/avs.go

```diff
@@ -37,25 +54,89 @@ func (k *Keeper) GetAVSMinimumSelfDelegation(ctx sdk.Context, avsAddr string) (s
 		return sdkmath.LegacyNewDec(0), errorsmod.Wrap(err, fmt.Sprintf("GetAVSMinimumSelfDelegation: key is %s", avsAddr))
 	}
 
-	return sdkmath.LegacyNewDec(avsInfo.Info.MinSelfDelegation.Int64()), nil
+	return sdkmath.LegacyNewDec(int64(avsInfo.Info.MinSelfDelegation)), nil
 }
 
 // GetEpochEndAVSs returns the AVS list where the current block marks the end of their epoch.
-func (k *Keeper) GetEpochEndAVSs(ctx sdk.Context) ([]string, error) {
-	var avsList []types.AVSInfo
-	k.IterateAVSInfo(ctx, func(_ int64, epochEndAVSInfo types.AVSInfo) (stop bool) {
-		avsList = append(avsList, epochEndAVSInfo)
+func (k *Keeper) GetEpochEndAVSs(ctx sdk.Context, epochIdentifier string, epochNumber int64) []string {
+	var avsList []string
+	k.IterateAVSInfo(ctx, func(_ int64, avsInfo types.AVSInfo) (stop bool) {
+		if epochIdentifier == avsInfo.EpochIdentifier && epochNumber > int64(avsInfo.StartingEpoch) {
+			avsList = append(avsList, avsInfo.AvsAddress)
+		}
+		return false
+	})
+
+	return avsList
+}
+
+// GetTaskChallengeEpochEndAVSs returns the AVS list where the current block marks the end of their epoch.
+func (k *Keeper) GetTaskChallengeEpochEndAVSs(ctx sdk.Context, epochIdentifier string, epochNumber int64) []string {
+	var avsList []string
+	k.IterateAVSInfo(ctx, func(_ int64, avsInfo types.AVSInfo) (stop bool) {
+		if epochIdentifier == avsInfo.EpochIdentifier && epochNumber > int64(avsInfo.StartingEpoch) {
+			avsList = append(avsList, avsInfo.AvsAddress)
+		}
 		return false
 	})
+	return avsList
+}
 
-	if len(avsList) == 0 {
-		return []string{}, nil
+func (k *Keeper) GetAVSAddrByChainID(ctx sdk.Context, chainID string) (string, error) {
+	chainID = ChainIDWithoutRevision(chainID)
+	if len(chainID) == 0 {
+		return "", errorsmod.Wrap(types.ErrNotNull, "RegisterAVSWithChainID: chainID is null")
+	}
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixAVSInfoByChainID)
+	if !store.Has([]byte(chainID)) {
+		return "", errorsmod.Wrap(types.ErrNoKeyInTheStore, fmt.Sprintf("GetAVSAddrByChainID: key is %s", chainID))
 	}
+	avsAddr := store.Get([]byte(chainID))
 
-	avsAddrList := make([]string, len(avsList))
-	for i := range avsList {
-		avsAddrList[i] = avsList[i].AvsAddress
+	return string(avsAddr), nil
+}
+
+func (k *Keeper) GetChainIDByAVSAddr(ctx sdk.Context, avsAddr string) (string, error) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixAVSInfoByChainID)
+	iterator := sdk.KVStorePrefixIterator(store, nil)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		chainID := string(iterator.Key())
+		if string(iterator.Value()) == avsAddr {
+			return chainID, nil
+		}
 	}
 
-	return avsAddrList, nil
+	return "", errorsmod.Wrap(types.ErrNoKeyInTheStore, fmt.Sprintf("GetChainIDByAVSAddr: key is %s", avsAddr))
+}
+
+// RegisterAVSWithChainID creates an avs address given the chainID
+func (k Keeper) RegisterAVSWithChainID(ctx sdk.Context, chainID string) (err error) {
```

Comment:
During this registration, the caller should be able to provide `AssetIDs`, `EpochsUntilUnbonded`, `EpochIdentifier`, `MinSelfDelegation` and `StartingEpoch`. Other values should be empty / blank, and we should call `AVsInfoUpdate`. This will ensure compatibility with all of the related AVS functions, like `GetEpochEndAVSs`, `GetAVSSupportedAssets`, and `GetAVSMinimumSelfDelegation`